File: kapptemplate/templates/C++/kapp4/src/%{APPNAMELC}view.cpp
/***************************************************************************
 *   Copyright (C) %{CURRENT_YEAR} by %{AUTHOR} <%{EMAIL}>                            *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This program is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program; if not, write to the                         *
 *   Free Software Foundation, Inc.,                                       *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA .        *
 ***************************************************************************/

#include "%{APPNAMELC}view.h"
#include "settings.h"

#include <KLocale>
#include <QtGui/QLabel>

%{APPNAME}View::%{APPNAME}View(QWidget *)
{
    ui_%{APPNAMELC}view_base.setupUi(this);
    settingsChanged();
    setAutoFillBackground(true);
}

%{APPNAME}View::~%{APPNAME}View()
{

}

void %{APPNAME}View::switchColors()
{
    // switch the foreground/background colors of the label
    QColor color = Settings::col_background();
    Settings::setCol_background( Settings::col_foreground() );
    Settings::setCol_foreground( color );

    settingsChanged();
}

void %{APPNAME}View::settingsChanged()
{
    QPalette pal;
    pal.setColor( QPalette::Window, Settings::col_background());
    pal.setColor( QPalette::WindowText, Settings::col_foreground());
    ui_%{APPNAMELC}view_base.kcfg_sillyLabel->setPalette( pal );

    // i18n : internationalization
    ui_%{APPNAMELC}view_base.kcfg_sillyLabel->setText( i18n("This project is %1 days old",Settings::val_time()) );
    emit signalChangeStatusbar( i18n("Settings changed") );
}

#include "%{APPNAMELC}view.moc"

File: kapptemplate/templates/ruby/kderubyapp/src/%{APPNAMELC}view.rb
require 'ui_%{APPNAMELC}view_base.rb'
require 'settings.rb'

class %{APPNAMEFU}View < Qt::Widget
  signals 'signalChangeStatusbar(QString)', 'signalChangeCaption(QString)'
  slots :switchColors, :settingsChanged

  attr_reader :settings

  def initialize( parent )
    super(parent)
    @ui = Ui_%{APPNAMEFU}view_base.new
    @ui.setupUi(self)
    # Note that the Settings class is a singleton
    @settings = Settings.instance
    settingsChanged()
    setAutoFillBackground(true)
  end

private
  def switchColors()
    # switch the foreground/background colors of the label
    color = @settings.col_background
    @settings.setCol_background( @settings.col_foreground )
    @settings.setCol_foreground( color )
    settingsChanged()
  end

  def settingsChanged()
    pal = Qt::Palette.new
    pal.setColor(Qt::Palette::Window, @settings.col_background)
    pal.setColor( Qt::Palette::WindowText, @settings.col_foreground)
    @ui.kcfg_sillyLabel.palette = pal
 
    # i18n : internationalization
    # @ui.kcfg_sillyLabel.text = KDE::i18n("This project is %%1 days old", @settings.val_time.to_s) 
    @ui.kcfg_sillyLabel.text = "This project is #{@settings.val_time} days old"
    emit signalChangeStatusbar( KDE::i18n("Settings changed") )
  end
end
